src/server/game/Achievements/AchievementMgr.cpp

```diff
@@ -439,6 +481,19 @@ void PlayerAchievementMgr::SendAchievementInfo(Player* receiver, uint32 /*achiev
         inspectedAchievements.Data.Earned.push_back(earned);
     }
 
+    for (auto itr = accountCriteriaProgress.begin(); itr != accountCriteriaProgress.end(); ++itr)
+    {
+        WorldPackets::Achievement::CriteriaProgress progress;
+        progress.Id = itr->first;
+        progress.Quantity = itr->second.Counter;
+        progress.Player = itr->second.PlayerGUID;
```

Comment:
iirc blizz sends bnet account guid here

Reply:
In SMSG_ALL_ACCOUNT_CRITERIA they send the BNetAccount guid and in SMSG_ALL_ACHIEVEMENT_DATA they send the player guid who earned the criteria

Reply:
then line 856 is wrong

Reply:
The PlayerGUID is a bit confusing and should have a more generic name since it can also be another type of guid. When the criteria is loaded from the db the PlayerGUID is set to the BNet account guid on line 790. But you are right. When earning a new criteria it's still filled with the player guid in CriteriaHandler line 954. This needs to be changed.